examples/raja-teams.cpp

```diff
@@ -0,0 +1,532 @@
+                       hipMemcpyHostToDevice ));
+   if(exec_place == RAJA::ExecPlace::DEVICE)
+    {
+      Ddat_ptr = Ddat_device;
+    }
+#endif
+
+    RAJA::View<int, RAJA::Layout<2>> D(Ddat_ptr, N_tri2, N_tri2);
+
+    RAJA::launch<launch_policy>( select_cpu_or_gpu,
+      RAJA::Resources(RAJA::Teams(N_tri2), RAJA::Threads(N_tri2)),
+        [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+
+          RAJA::loop<teams_x>(ctx, RAJA::RangeSegment(0, N_tri), [&](int r) {
+
+            RAJA::loop<threads_x>(ctx, RAJA::RangeSegment(r, N_tri), [&](int c) {
+              D(r, c) = r * N_tri + c;
+            });  // loop j
+
+            RAJA::loop<threads_x>(ctx, RAJA::RangeSegment(r, N_tri), [&](int c) {
+              printf("r=%d, c=%d : D=%d\n", r, c, D(r, c));
+            });  // loop c
+          });    // loop r
+
+        });      // outer lambda
+
+    memoryManager::deallocate(Ddat);
+
+#if defined(RAJA_ENABLE_HIP)
+    memoryManager::deallocate_gpu(Ddat_device);
+#endif
+
+
+    //========================
+    // Matrix-Matrix Multiplication Example
+    //========================
+    std::cout << "\n Running Matrix-Matrix Multiplication example...\n";
+
+    double *A_ptr=A;
+    double *B_ptr=B;
+    double *C_ptr=C;
+#if defined(RAJA_ENABLE_HIP)
+    double* A_device = memoryManager::allocate_gpu<double>(N*N);
+    double* B_device = memoryManager::allocate_gpu<double>(N*N);
+    double* C_device = memoryManager::allocate_gpu<double>(N*N);
+    hipErrchk(hipMemcpy( A_device, A, N * N * sizeof(double), hipMemcpyHostToDevice ));
+    hipErrchk(hipMemcpy( B_device, B, N * N * sizeof(double), hipMemcpyHostToDevice ));
+    hipErrchk(hipMemcpy( C_device, C, N * N * sizeof(double), hipMemcpyHostToDevice ));
+
+   if(exec_place == RAJA::ExecPlace::DEVICE)
+    {
+      A_ptr=A_device;
+      B_ptr=B_device;
+      C_ptr=C_device;
+    }
+#endif
+
+    // Set up Teams/Threads
+
+    printf("select_cpu_or_gpu %d \n", select_cpu_or_gpu);
+    RAJA::launch<launch_policy>(select_cpu_or_gpu,
+        RAJA::Resources(RAJA::Teams(NBlocks, NBlocks),
+                        RAJA::Threads(NThreads, NThreads)),
+        [=] RAJA_HOST_DEVICE(RAJA::LaunchContext ctx) {
+
+          RAJA::RangeSegment TeamRange(0, NBlocks);
+          RAJA::RangeSegment ThreadRange(0, NThreads);
+          //
+          // Loop over teams
+          //
+          // RAJA::loop<teams1>(ctx, RAJA::RangeSegment(0, NBlocks), [&](int by)
+          // { RAJA::loop<teams_x>(ctx, RAJA::RangeSegment(0, NBlocks), [&](int
+          // bx) {
+          RAJA::loop<teams_xy>(ctx, TeamRange, TeamRange, [&](int bx, int by) {
+
+            TEAM_SHARED double As[THREAD_BLOCK_SZ][THREAD_BLOCK_SZ];
+            TEAM_SHARED double Bs[THREAD_BLOCK_SZ][THREAD_BLOCK_SZ];
+            TEAM_SHARED double Cs[THREAD_BLOCK_SZ][THREAD_BLOCK_SZ];
+
+            // Team parallel loop
+            RAJA::loop<threads_y>(ctx, ThreadRange, [&](int ty) {
```

Comment:
@ajkunen Currently the RAJA::LaunchContext doesn't do anything inside RAJA::loop, thoughts on removing it?